src/main/kotlin/com/avail/descriptor/representation/AbstractDescriptor.kt

```diff
@@ -0,0 +1,4202 @@
+		private val bitFieldsCache: MutableMap<
+			IntegerSlotsEnum,
+			List<BitField>
+		> = HashMap(500)
+
+		/** A [ReadWriteLock] that protects the [bitFieldsCache]. */
+		private val bitFieldsLock: ReadWriteLock = ReentrantReadWriteLock()
+
+		/**
+		 * Describe the integer field onto the provided [StringBuilder]. The
+		 * pre-extracted `long` value is provided, as well as the containing
+		 * [AvailObject] and the [IntegerSlotsEnum] instance. Take into account
+		 * annotations on the slot enumeration object which may define the way
+		 * it should be described.
+		 *
+		 * @param self
+		 *   The object containing the `int` value in some slot.
+		 * @param value
+		 *   The `long` value of the slot.
+		 * @param slot
+		 *   The [integer slot][IntegerSlotsEnum] definition.
+		 * @param bitFields
+		 *   The slot's [BitField]s, if any.
+		 * @param builder
+		 *   Where to write the description.
+		 */
+		fun describeIntegerSlot (
+			self: AvailObject,
+			value: Long,
+			slot: IntegerSlotsEnum,
+			bitFields: List<BitField>,
+			builder: StringBuilder)
+		{
+			try
+			{
+				val slotName = slot.fieldName()
+				if (bitFields.isEmpty())
+				{
+					val slotMirror = slot.javaClass.getField(slotName)
+					val enumAnnotation =
+						slotMirror.getAnnotation(EnumField::class.java)
+					var numBits = 64
+					if (enumAnnotation != null)
+					{
+						val enumClass = enumAnnotation.describedBy.java
+						val enumValues = enumClass.enumConstants
+						numBits = 64 - java.lang.Long.numberOfLeadingZeros(
+							enumValues.size.toLong())
+					}
+					builder.append(" = ")
+					describeIntegerField(
+						value, numBits, enumAnnotation, builder)
+				}
+				else
+				{
+					builder.append("(")
+					var first = true
+					for (bitField in bitFields)
+					{
+						if (!first)
+						{
+							builder.append(", ")
+						}
+						builder.append(bitField.name)
+						builder.append("=")
+						describeIntegerField(
+							self.slot(bitField).toLong(),
+							bitField.bits,
+							bitField.enumField,
+							builder)
+						first = false
+					}
+					builder.append(")")
+				}
+			}
+			catch (e: SecurityException)
+			{
+				throw RuntimeException(e)
+			}
+			catch (e: IllegalArgumentException)
+			{
+				throw RuntimeException(e)
+			}
+			catch (e: ReflectiveOperationException)
+			{
+				throw RuntimeException(e)
+			}
+		}
+
+		/**
+		 * Extract the [integer slot][IntegerSlotsEnum]'s [List] of [BitField]s,
+		 * excluding ones marked with the annotation @[HideFieldInDebugger].
+		 *
+		 * @param slot
+		 *   The integer slot.
+		 * @return
+		 *   The slot's bit fields.
+		 */
+		fun bitFieldsFor (slot: IntegerSlotsEnum): List<BitField>
+		{
+			bitFieldsLock.readLock().lock()
+			try
+			{
+				// Vast majority of cases.
+				val bitFields = bitFieldsCache[slot]
+				if (bitFields != null)
+				{
+					return bitFields
+				}
+			}
+			finally
+			{
+				bitFieldsLock.readLock().unlock()
+			}
+			bitFieldsLock.writeLock().lock()
```

Comment:
`return bitFieldsLock.write {`

Reply:
Done. I fixed this everywhere. There were three occurrences in Kotlin.